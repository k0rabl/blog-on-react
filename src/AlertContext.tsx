import React, { Component } from 'react'

interface IState {
  isOpen: boolean
  type: string
}

const AlertContext = React.createContext({
    isOpen: false,
    type: '',
    setOpen: (c: boolean) => {},
    setType: (c: string) => {}
  })


class AlertProvider extends Component {
  // Context state
  state: IState = {
    isOpen: false,
    type: ''
  }
 

  // Method to update state
  setOpen = (isOpen: boolean) => {
    this.setState((prevState) => ({ isOpen }))
  }

  setType = (type: string) => {
    this.setState((prevState) => ({ type }))
  }

  render() {
    const { children } = this.props
    const { isOpen, type } = this.state
    const { setType, setOpen } = this

    return (
      <AlertContext.Provider
        value={{
          isOpen,
          type,
          setOpen,
          setType,
        }}
      >
        {children}
      </AlertContext.Provider>
    )
  }
}

export default AlertContext

export { AlertProvider }