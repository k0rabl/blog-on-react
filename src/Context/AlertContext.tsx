import React, { Component } from 'react' 

interface IState {
  isOpen: boolean
  message: string
}

const AlertContext = React.createContext({
    isOpen: false,
    message: '',
    setOpen: (c: boolean) => {},
    setType: (c: string) => {}
  })


class AlertProvider extends Component {
  // Context state
  state: IState = {
    isOpen: false,
    message: ''
  }
 

  // Method to update state
  setOpen = (isOpen: boolean) => {
    this.setState((prevState) => ({ isOpen }))
  }

  setType = (message: string) => {
    this.setState((prevState) => ({ message }))
  }

  render() {
    const { children } = this.props
    const { isOpen, message } = this.state
    const { setType, setOpen } = this

    return (
      <AlertContext.Provider
        value={{
          isOpen,
          message,
          setOpen,
          setType,
        }}
      >
        {children}
      </AlertContext.Provider>
    )
  }
}

export default AlertContext

export { AlertProvider }