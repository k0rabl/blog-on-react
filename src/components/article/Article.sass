.article
  margin-top: 25px
  margin-bottom: 25px
  .top
    margin-bottom: 15px
    display: flex
    justify-content: space-between
    .title
      font-size: 26px