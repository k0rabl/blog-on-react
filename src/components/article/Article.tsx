import React, { Component } from 'react'
import IArticle from '../../interfaces/IArticle'

class Article extends Component<IArticle> {
  constructor(props: IArticle){
    super(props)
  }

  render() {
    const { date, name, desc } = this.props
    
    return (
      <div className="container">
        <div className="top">
          <div className="title">{name}</div>
          <div className="date">{date.toDateString()}</div>
        </div>
        <div className="bottom">
          <div className="image"></div>
          <div className="descr">{desc}</div>
        </div>
      </div>
    )
  }
}

export default Article