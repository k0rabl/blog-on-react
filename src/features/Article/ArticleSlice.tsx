import { createSlice, PayloadAction } from '@reduxjs/toolkit'
import Articles from '../../fixtures/Articles'
import IArticle from './component/IArticle'

export interface PaginationState {
  filteredArticles: IArticle[]
  articles: IArticle[]
}

const initialState: PaginationState = {
  filteredArticles: Articles,
  articles: Articles
}

export const articlesSlice = createSlice({
  name: 'search',
  initialState,
  reducers: {
    handleRead: (state, action: PayloadAction<number>) => {
      state.articles =  state.articles.map((element: IArticle) => ({
        ...element,
        isRead: element.id === action.payload ? true : element.isRead
      }))
    },
    handleDeleteElement: (state, action: PayloadAction<number>) => {
      state.articles = state.articles.filter(element => {
        return element.id !== action.payload
      })
    },
    handleEditElement: (state, action: PayloadAction<IArticle>) => {
      state.articles = state.articles.map((element: IArticle) => {
        if(element.id === action.payload.id)  
          return action.payload

        return element
      })
    },
    handleAddElement: (state, action: PayloadAction<IArticle>) => {
      state.articles.push(action.payload)
    }
  }
})

export const { 
  handleRead, 
  handleDeleteElement,
  handleEditElement,
  handleAddElement
} = articlesSlice.actions
export default articlesSlice.reducer