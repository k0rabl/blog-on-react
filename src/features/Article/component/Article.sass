
.article
  margin-top: 25px
  margin-bottom: 25px
  width: 100%
  display: flex
  justify-content: space-between
  .leftImage
    width: 30%
    border: 1px solid #b10303
    margin-right: 15px
    display: flex
    justify-content: center
    align-items: center
    img
      width: 100%
      height: 200px
  .rightDescr
    width: 70%
  &Link
    color: inherit
  
  .top
    margin-bottom: 15px
    display: flex
    justify-content: space-between
    &-left
      display: flex
      align-items: center
      .title
        margin-right: 15px
        font-size: 26px
        color: #b10303
        transition: color .2s
        &.edit
          color: #B94A3E
          white-space: nowrap
          text-overflow: ellipsis
          max-width: 350px
          overflow: hidden
        &.read
          color: #B94A3E
        &:hover
          color: #742c24
      .btn
        &-delete
          margin-right: 15px