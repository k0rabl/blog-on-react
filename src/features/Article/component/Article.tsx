import React, { Component } from 'react'
import IArticle from './IArticle'

import './Article.sass'
import { Link } from 'react-router-dom'

class Article extends Component<IArticle> {

  render() {
    const { id, date, name, desc, isRead } = this.props
    
    return (
      <Link to={`/${id}`} className="link articleLink">
        <div className="container article">
          <div className="top">
            <div className={`title ${isRead && 'read'}`}>{name}</div>
            <div className="date">{date}</div>
          </div>
          <div className="bottom">
            <div className="image"></div>
            <div className="descr">{desc}</div>
          </div>
        </div>
      </Link>
    )
  }
}

export default Article