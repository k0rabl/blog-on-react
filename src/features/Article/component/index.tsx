import React, { PureComponent } from 'react'
import { ArticleProps, IState } from './IArticle'

import './Article.sass'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import { RootState } from '../../../redux/store'
import { handleDeleteElement } from '../ArticleSlice'
import { Modal } from '../../Modal'
import AlertContext from '../../../AlertContext'

class Article extends PureComponent<ArticleProps, IState> {
  static contextType = AlertContext

  constructor(props: ArticleProps){
    super(props)

    this.state = {
      opened: false
    }
  }

  handleDelete = () => {
    const { setOpen, setType } = this.context 
    const { id, handleDeleteElement } = this.props
    
    handleDeleteElement(Number(id))
    
    setOpen(true)
    setType('Delete')
    this.setState({opened: false})
  }



  handleShowModal = () => {
    const { opened } = this.state
    this.setState({opened: !opened})
  }

  render() {
    const { 
      id, 
      date, 
      name, 
      desc, 
      isRead,
      editMode
    } = this.props   
    
    return (
      <div className="container article">
        <div className="top">
          <div className="top-left">
            
            <Link to={`/${id}`} className="link articleLink">
              <div className={`title ${isRead && 'read'} ${editMode && 'edit'}`}>{name}</div>
            </Link>
            
            {editMode && 
              <div className="buttons">
                <button  className="waves-effect waves-light redC-bg lighten-3 btn btn-delete" onClick={this.handleShowModal}>
                  <i className="material-icons center">delete</i>
                </button > 
                
                {/* <button  className="waves-effect waves-light redC-bg lighten-3 btn btn-edit" onClick={() => console.log('edit')}>
                  <i className="material-icons center">edit</i>
                </button > */} 
              </div>
            }
          </div>
          
          <div className="date">{date}</div>
        </div>
        <div className="bottom">
          <div className="image"></div>
          <div className="descr">{desc}</div>
        </div>
        <Modal 
          opened={this.state.opened}
          onClose={this.handleShowModal} 
          onAccept={this.handleDelete} 
          type='delete'
        />
      </div>
    )
  }
}


const mapStateToProps = (state: RootState) => ({
  editMode: state.edit.editMode,
  articles: state.search.filteredArticles
})

const mapDispatch = { handleDeleteElement }

export default connect(mapStateToProps, mapDispatch)(Article)