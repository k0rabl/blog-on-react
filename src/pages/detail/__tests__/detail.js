// import 'jsdom-global/register'
import React from "react"
import ConnectedDetail, { Detail } from '../index'
import { mount, render } from 'enzyme';

import {
  Router,
  Route
} from "react-router-dom"

import {createMemoryHistory} from 'history'

import configureStore from 'redux-mock-store'
import { Provider } from "react-redux";
import { handleRead, handleEditElement, handleAddElement } from '../../../features/Article/ArticleSlice'

import Articles from "../../../fixtures/Articles";
import { AlertProvider } from "../../../context/AlertContext";

// const history = createMemoryHistory()

// function renderWithRouterMatch(
//   ui,
//   {
//     path = "/",
//     route = "/",
//     history = createMemoryHistory({ initialEntries: [route] })
//   } = {}
// ) {
//   const mockStore = configureStore();

//   const store = mockStore({
//     search: {        
//       articles: Articles,
//     },
//     edit: {        
//       editMode: true,
//     }
//   })
//   return {
//     ...mount(
      
//       <Provider store={store}>
//         <AlertProvider>
//           <Router history={history}>
//             <Route path={path} component={ui} />
//           </Router>
//         </AlertProvider>
//       </Provider>
//     )
//   };
// }

describe('<Detail />', () => {
  const initialState = { 
    search: {        
      articles: Articles,
    },
    edit: {        
      editMode: true,
    }
  };
  const mockStore = configureStore();
  let store, component
  
  const history = createMemoryHistory({ initialEntries: ["/1"] })

  beforeEach(() => {
    store = mockStore(initialState)

    component = mount(
      <Provider store={store}>
        <AlertProvider>
          <Router history={history} >
            <Route path="/:id" component={ConnectedDetail} />
          </Router> 
        </AlertProvider>
      </Provider>
    )
  })

  it('+++ render the component', () => {
    // expect(component.find(ConnectedDetail).length).toEqual(1)
    // renderWithRouterMatch(ConnectedDetail, {
    //   route: "/1",
    //   path: "/:id"
    // });
    expect(component).toMatchSnapshot()
  })

  
  it('+++ render witchout id', () => {
    store = mockStore(initialState)

    component = mount(
      <Provider store={store}>
        <AlertProvider>
          <Router history={history} >
            <ConnectedDetail />
          </Router> 
        </AlertProvider>
      </Provider>
    )
    
    expect(component).toMatchSnapshot()
  })

  it('+++ check Button back', () => {
    component
      .find('button.button-back')
      .simulate('click')
  })

  it('+++ check Button save', () => {
    component
      .find('button.button-save')
      .simulate('click')
      
  })

  it('+++ should base64 file encoding', () => {
    const file = new File([""], 'darthvader.png', {type: 'image/*'});

    component.find('input#file').simulate('change', {
      target: {
        files: [file]
      }
    })
  })
  
  it('+++ should fet errror size', () => {
    const file = new File([""], 'darthvader.png', {type: 'image/*'})
    Object.defineProperty(file, 'size', { value: 1024 * 5024 + 1 })

    component.find('input#file').simulate('change', {
      target: {
        files: [file]
      }
    })       
  })

  it('+++ should change input', () => {

    component.find('input.input__name').simulate('change', {
      target: {
        value: 'test'
      }
    })
  })

  it('+++ check Prop matches with initialState', () => {
    expect(component.find(Detail).prop('articles')).toEqual(initialState.search.articles)
      
  })

  it('+++ check actions on dispatching ', () => {
    let action
    store = mockStore(initialState)

    component = mount(
      <Provider store={store}>
        <AlertProvider>
          <Router history={history} >
            <ConnectedDetail />
          </Router> 
        </AlertProvider>
      </Provider>
    )

    store.dispatch(handleRead(true))
    store.dispatch(handleEditElement(2))
    store.dispatch(handleAddElement({
      id: 1,
      date: '2021-05-20',
      name: 'Test name',
      image: '',
      desc: 'test-1.',
      isRead: true
    }))

    action = store.getActions()
    
    expect(action[0].type).toBe("search/handleRead")
    expect(action[1].type).toBe("search/handleEditElement")
    expect(action[2].type).toBe("search/handleAddElement")
  })
})