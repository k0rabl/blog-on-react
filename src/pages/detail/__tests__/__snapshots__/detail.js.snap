// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Detail /> +++ render the component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <AlertProvider>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "0wux6n",
              "pathname": "/1",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "0wux6n",
            "pathname": "/1",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Route
        component={[Function]}
        path="/:id"
      >
        <withRouter(Connect(Detail))
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "canGo": [Function],
              "createHref": [Function],
              "entries": Array [
                Object {
                  "hash": "",
                  "key": "0wux6n",
                  "pathname": "/1",
                  "search": "",
                  "state": undefined,
                },
              ],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "index": 0,
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "key": "0wux6n",
                "pathname": "/1",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          location={
            Object {
              "hash": "",
              "key": "0wux6n",
              "pathname": "/1",
              "search": "",
              "state": undefined,
            }
          }
          match={
            Object {
              "isExact": true,
              "params": Object {
                "id": "1",
              },
              "path": "/:id",
              "url": "/1",
            }
          }
        >
          <Connect(Detail)
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "canGo": [Function],
                "createHref": [Function],
                "entries": Array [
                  Object {
                    "hash": "",
                    "key": "0wux6n",
                    "pathname": "/1",
                    "search": "",
                    "state": undefined,
                  },
                ],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "index": 0,
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "key": "0wux6n",
                  "pathname": "/1",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "key": "0wux6n",
                "pathname": "/1",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {
                  "id": "1",
                },
                "path": "/:id",
                "url": "/1",
              }
            }
          >
            <Detail
              articles={
                Array [
                  Object {
                    "date": "2021-05-20",
                    "desc": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusantium id deleniti suscipit nisi mollitia aliquid expedita earum voluptates excepturi, quod ratione ut a, laboriosam tempore blanditiis quasi aspernatur est adipisci tenetur quis vero voluptate fugiat. Minus necessitatibus fugiat non at, fuga quibusdam tenetur ratione quasi numquam vel explicabo nesciunt ipsum.",
                    "id": 1,
                    "image": "",
                    "isRead": true,
                    "name": "How to use hooks in React?",
                    "preview": "Lorem ipsum dolor sit amet consectetur adipisicing elit.",
                  },
                  Object {
                    "date": "2021-05-20",
                    "desc": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusantium id deleniti suscipit nisi mollitia aliquid expedita earum voluptates excepturi, quod ratione ut a, laboriosam tempore blanditiis quasi aspernatur est adipisci tenetur quis vero voluptate fugiat. Minus necessitatibus fugiat non at, fuga quibusdam tenetur ratione quasi numquam vel explicabo nesciunt ipsum.",
                    "id": 2,
                    "image": "",
                    "isRead": false,
                    "name": "How to use hooks in React? (||)",
                  },
                  Object {
                    "date": "2021-05-20",
                    "desc": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusantium id deleniti suscipit nisi mollitia aliquid expedita earum voluptates excepturi, quod ratione ut a, laboriosam tempore blanditiis quasi aspernatur est adipisci tenetur quis vero voluptate fugiat. Minus necessitatibus fugiat non at, fuga quibusdam tenetur ratione quasi numquam vel explicabo nesciunt ipsum.",
                    "id": 3,
                    "image": "",
                    "isRead": false,
                    "name": "How to use hooks in React?",
                  },
                  Object {
                    "date": "2021-04-30",
                    "desc": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusantium id deleniti suscipit nisi mollitia aliquid expedita earum voluptates excepturi, quod ratione ut a, laboriosam tempore blanditiis quasi aspernatur est adipisci tenetur quis vero voluptate fugiat. Minus necessitatibus fugiat non at, fuga quibusdam tenetur ratione quasi numquam vel explicabo nesciunt ipsum.",
                    "id": 4,
                    "image": "",
                    "isRead": false,
                    "name": "How to use hooks in React?",
                  },
                  Object {
                    "date": "2021-11-14",
                    "desc": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusantium id deleniti suscipit nisi mollitia aliquid expedita earum voluptates excepturi, quod ratione ut a, laboriosam tempore blanditiis quasi aspernatur est adipisci tenetur quis vero voluptate fugiat. Minus necessitatibus fugiat non at, fuga quibusdam tenetur ratione quasi numquam vel explicabo nesciunt ipsum.",
                    "id": 5,
                    "image": "",
                    "isRead": false,
                    "name": "How to use hooks in React?",
                  },
                  Object {
                    "date": "2021-05-20",
                    "desc": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusantium id deleniti suscipit nisi mollitia aliquid expedita earum voluptates excepturi, quod ratione ut a, laboriosam tempore blanditiis quasi aspernatur est adipisci tenetur quis vero voluptate fugiat. Minus necessitatibus fugiat non at, fuga quibusdam tenetur ratione quasi numquam vel explicabo nesciunt ipsum.",
                    "id": 6,
                    "image": "",
                    "isRead": false,
                    "name": "second page",
                  },
                  Object {
                    "date": "2021-05-20",
                    "desc": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusantium id deleniti suscipit nisi mollitia aliquid expedita earum voluptates excepturi, quod ratione ut a, laboriosam tempore blanditiis quasi aspernatur est adipisci tenetur quis vero voluptate fugiat. Minus necessitatibus fugiat non at, fuga quibusdam tenetur ratione quasi numquam vel explicabo nesciunt ipsum.",
                    "id": 7,
                    "image": "",
                    "isRead": false,
                    "name": "How to use hooks in React?",
                  },
                  Object {
                    "date": "2021-05-20",
                    "desc": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusantium id deleniti suscipit nisi mollitia aliquid expedita earum voluptates excepturi, quod ratione ut a, laboriosam tempore blanditiis quasi aspernatur est adipisci tenetur quis vero voluptate fugiat. Minus necessitatibus fugiat non at, fuga quibusdam tenetur ratione quasi numquam vel explicabo nesciunt ipsum.",
                    "id": 8,
                    "image": "",
                    "isRead": false,
                    "name": "How to use hooks in React?",
                  },
                  Object {
                    "date": "2021-05-20",
                    "desc": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusantium id deleniti suscipit nisi mollitia aliquid expedita earum voluptates excepturi, quod ratione ut a, laboriosam tempore blanditiis quasi aspernatur est adipisci tenetur quis vero voluptate fugiat. Minus necessitatibus fugiat non at, fuga quibusdam tenetur ratione quasi numquam vel explicabo nesciunt ipsum.",
                    "id": 9,
                    "image": "",
                    "isRead": false,
                    "name": "How to use hooks in React?",
                  },
                ]
              }
              editMode={true}
              handleAddElement={[Function]}
              handleEditElement={[Function]}
              handleRead={[Function]}
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "canGo": [Function],
                  "createHref": [Function],
                  "entries": Array [
                    Object {
                      "hash": "",
                      "key": "0wux6n",
                      "pathname": "/1",
                      "search": "",
                      "state": undefined,
                    },
                  ],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "index": 0,
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "key": "0wux6n",
                    "pathname": "/1",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              location={
                Object {
                  "hash": "",
                  "key": "0wux6n",
                  "pathname": "/1",
                  "search": "",
                  "state": undefined,
                }
              }
              match={
                Object {
                  "isExact": true,
                  "params": Object {
                    "id": "1",
                  },
                  "path": "/:id",
                  "url": "/1",
                }
              }
            >
              <div
                className="detail"
              >
                <div
                  className="buttons"
                >
                  <button
                    className="button button-back"
                    onClick={[Function]}
                  >
                    Back
                  </button>
                  <button
                    className="button button-save"
                    onClick={[Function]}
                  >
                    Save
                  </button>
                </div>
                <div
                  className="addImage"
                >
                  <input
                    accept="image/*"
                    id="file"
                    name="image"
                    onChange={[Function]}
                    type="file"
                  />
                  <img
                    alt=""
                    src=""
                  />
                </div>
                <input
                  className="input input__name"
                  name="name"
                  onChange={[Function]}
                  type="text"
                  value="How to use hooks in React?"
                />
                <textarea
                  className="input input__desc"
                  name="desc"
                  onChange={[Function]}
                  value="Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusantium id deleniti suscipit nisi mollitia aliquid expedita earum voluptates excepturi, quod ratione ut a, laboriosam tempore blanditiis quasi aspernatur est adipisci tenetur quis vero voluptate fugiat. Minus necessitatibus fugiat non at, fuga quibusdam tenetur ratione quasi numquam vel explicabo nesciunt ipsum."
                />
                <p
                  className="date"
                >
                  2021-05-20
                </p>
              </div>
            </Detail>
          </Connect(Detail)>
        </withRouter(Connect(Detail))>
      </Route>
    </Router>
  </AlertProvider>
</Provider>
`;

exports[`<Detail /> +++ render witchout id 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <AlertProvider>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "0wux6n",
              "pathname": "/1",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "0wux6n",
            "pathname": "/1",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <withRouter(Connect(Detail))>
        <Connect(Detail)
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "canGo": [Function],
              "createHref": [Function],
              "entries": Array [
                Object {
                  "hash": "",
                  "key": "0wux6n",
                  "pathname": "/1",
                  "search": "",
                  "state": undefined,
                },
              ],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "index": 0,
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "key": "0wux6n",
                "pathname": "/1",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          location={
            Object {
              "hash": "",
              "key": "0wux6n",
              "pathname": "/1",
              "search": "",
              "state": undefined,
            }
          }
          match={
            Object {
              "isExact": false,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
        >
          <Detail
            articles={
              Array [
                Object {
                  "date": "2021-05-20",
                  "desc": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusantium id deleniti suscipit nisi mollitia aliquid expedita earum voluptates excepturi, quod ratione ut a, laboriosam tempore blanditiis quasi aspernatur est adipisci tenetur quis vero voluptate fugiat. Minus necessitatibus fugiat non at, fuga quibusdam tenetur ratione quasi numquam vel explicabo nesciunt ipsum.",
                  "id": 1,
                  "image": "",
                  "isRead": true,
                  "name": "How to use hooks in React?",
                  "preview": "Lorem ipsum dolor sit amet consectetur adipisicing elit.",
                },
                Object {
                  "date": "2021-05-20",
                  "desc": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusantium id deleniti suscipit nisi mollitia aliquid expedita earum voluptates excepturi, quod ratione ut a, laboriosam tempore blanditiis quasi aspernatur est adipisci tenetur quis vero voluptate fugiat. Minus necessitatibus fugiat non at, fuga quibusdam tenetur ratione quasi numquam vel explicabo nesciunt ipsum.",
                  "id": 2,
                  "image": "",
                  "isRead": false,
                  "name": "How to use hooks in React? (||)",
                },
                Object {
                  "date": "2021-05-20",
                  "desc": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusantium id deleniti suscipit nisi mollitia aliquid expedita earum voluptates excepturi, quod ratione ut a, laboriosam tempore blanditiis quasi aspernatur est adipisci tenetur quis vero voluptate fugiat. Minus necessitatibus fugiat non at, fuga quibusdam tenetur ratione quasi numquam vel explicabo nesciunt ipsum.",
                  "id": 3,
                  "image": "",
                  "isRead": false,
                  "name": "How to use hooks in React?",
                },
                Object {
                  "date": "2021-04-30",
                  "desc": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusantium id deleniti suscipit nisi mollitia aliquid expedita earum voluptates excepturi, quod ratione ut a, laboriosam tempore blanditiis quasi aspernatur est adipisci tenetur quis vero voluptate fugiat. Minus necessitatibus fugiat non at, fuga quibusdam tenetur ratione quasi numquam vel explicabo nesciunt ipsum.",
                  "id": 4,
                  "image": "",
                  "isRead": false,
                  "name": "How to use hooks in React?",
                },
                Object {
                  "date": "2021-11-14",
                  "desc": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusantium id deleniti suscipit nisi mollitia aliquid expedita earum voluptates excepturi, quod ratione ut a, laboriosam tempore blanditiis quasi aspernatur est adipisci tenetur quis vero voluptate fugiat. Minus necessitatibus fugiat non at, fuga quibusdam tenetur ratione quasi numquam vel explicabo nesciunt ipsum.",
                  "id": 5,
                  "image": "",
                  "isRead": false,
                  "name": "How to use hooks in React?",
                },
                Object {
                  "date": "2021-05-20",
                  "desc": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusantium id deleniti suscipit nisi mollitia aliquid expedita earum voluptates excepturi, quod ratione ut a, laboriosam tempore blanditiis quasi aspernatur est adipisci tenetur quis vero voluptate fugiat. Minus necessitatibus fugiat non at, fuga quibusdam tenetur ratione quasi numquam vel explicabo nesciunt ipsum.",
                  "id": 6,
                  "image": "",
                  "isRead": false,
                  "name": "second page",
                },
                Object {
                  "date": "2021-05-20",
                  "desc": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusantium id deleniti suscipit nisi mollitia aliquid expedita earum voluptates excepturi, quod ratione ut a, laboriosam tempore blanditiis quasi aspernatur est adipisci tenetur quis vero voluptate fugiat. Minus necessitatibus fugiat non at, fuga quibusdam tenetur ratione quasi numquam vel explicabo nesciunt ipsum.",
                  "id": 7,
                  "image": "",
                  "isRead": false,
                  "name": "How to use hooks in React?",
                },
                Object {
                  "date": "2021-05-20",
                  "desc": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusantium id deleniti suscipit nisi mollitia aliquid expedita earum voluptates excepturi, quod ratione ut a, laboriosam tempore blanditiis quasi aspernatur est adipisci tenetur quis vero voluptate fugiat. Minus necessitatibus fugiat non at, fuga quibusdam tenetur ratione quasi numquam vel explicabo nesciunt ipsum.",
                  "id": 8,
                  "image": "",
                  "isRead": false,
                  "name": "How to use hooks in React?",
                },
                Object {
                  "date": "2021-05-20",
                  "desc": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusantium id deleniti suscipit nisi mollitia aliquid expedita earum voluptates excepturi, quod ratione ut a, laboriosam tempore blanditiis quasi aspernatur est adipisci tenetur quis vero voluptate fugiat. Minus necessitatibus fugiat non at, fuga quibusdam tenetur ratione quasi numquam vel explicabo nesciunt ipsum.",
                  "id": 9,
                  "image": "",
                  "isRead": false,
                  "name": "How to use hooks in React?",
                },
              ]
            }
            editMode={true}
            handleAddElement={[Function]}
            handleEditElement={[Function]}
            handleRead={[Function]}
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "canGo": [Function],
                "createHref": [Function],
                "entries": Array [
                  Object {
                    "hash": "",
                    "key": "0wux6n",
                    "pathname": "/1",
                    "search": "",
                    "state": undefined,
                  },
                ],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "index": 0,
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "key": "0wux6n",
                  "pathname": "/1",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "key": "0wux6n",
                "pathname": "/1",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": false,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
          >
            <div
              className="detail"
            >
              <div
                className="buttons"
              >
                <button
                  className="button button-back"
                  onClick={[Function]}
                >
                  Back
                </button>
                <button
                  className="button button-save"
                  onClick={[Function]}
                >
                  Save
                </button>
              </div>
              <div
                className="addImage"
              >
                <input
                  accept="image/*"
                  id="file"
                  name="image"
                  onChange={[Function]}
                  type="file"
                />
                <img
                  alt=""
                  src=""
                />
              </div>
              <input
                className="input input__name"
                name="name"
                onChange={[Function]}
                type="text"
                value=""
              />
              <textarea
                className="input input__desc"
                name="desc"
                onChange={[Function]}
                value=""
              />
              <p
                className="date"
              >
                2021-12-23
              </p>
            </div>
          </Detail>
        </Connect(Detail)>
      </withRouter(Connect(Detail))>
    </Router>
  </AlertProvider>
</Provider>
`;
